Regardless if the process runs in foregrund or background, as a
courtesy to users, we should always create the PID file.  Many
modern init run daemons in the foreground to monitor them, and
some use the creation of PID file as a sign to start any other
process that may depend on it.

diff -ruN busybox-1.34.1.orig/networking/ntpd.c busybox-1.34.1/networking/ntpd.c
--- busybox-1.34.1.orig/networking/ntpd.c	2021-04-26 19:21:10.000000000 +0200
+++ busybox-1.34.1/networking/ntpd.c	2022-01-16 13:53:08.885516777 +0100
@@ -2431,9 +2431,6 @@
 		G.stratum = 1;
 	}
 
-	if (!(opts & OPT_n)) /* only if backgrounded: */
-		write_pidfile_std_path_and_ext("ntpd");
-
 	/* If network is up, syncronization occurs in ~10 seconds.
 	 * We give "ntpd -q" 10 seconds to get first reply,
 	 * then another 50 seconds to finish syncing.
@@ -2460,6 +2457,8 @@
 		, SIG_IGN
 	);
 //TODO: free unused elements of key_entries?
+
+	write_pidfile_std_path_and_ext("ntpd");
 }
 
 int ntpd_main(int argc UNUSED_PARAM, char **argv) MAIN_EXTERNALLY_VISIBLE;
